pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -541,12 +547,23 @@ func isSecretValid(existingSecret v1.Secret) bool {
 	}
 
 	// if the calculated hash is different from the calculation, then it's invalid
-	if existingSecret.Annotations[esv1beta1.AnnotationDataHash] != utils.ObjectHash(existingSecret.Data) {
+	if existingSecret.Annotations[esv1beta1.AnnotationDataHash] != hashData(existingSecret.Data) {
```

Comment:
This does not seem correct to me. In the PR `existingSecret.Annotations[esv1beta1.AnnotationDataHash]` was computed by taking a map of secret data but it compares data directly.

Reply:
The value of the annotation is equivalent to the merge of existing data + new data.
Previously, the annotation value was equal to new data only. 

Then the secret is updated.

On the next iteration, the value of the annotation will be equal to the has of the current data.

Reply:
Ah, sorry you are right!